app/models/form526_submission.rb

```diff
@@ -126,9 +126,12 @@ def submit_with_birls_id_that_hasnt_been_tried_yet!(
     log_exception_to_sentry e, extra_content_for_sentry
   end
 
+  # Note that the User record is stored in Redis temporarily -- `User.redis_namespace_ttl`
   def get_first_name
     user = User.find(user_uuid)
-    user&.first_name&.upcase
+    return user&.first_name&.upcase if user&.first_name.present?
+
+    auth_headers&.dig('va_eauth_firstName')&.upcase
```

Comment:
Another way to write this:

```suggestion
    user&.first_name&.upcase.presence ||
      auth_headers&.dig('va_eauth_firstName')&.upcase
```